### ISSUE TITLE
--match should consider only basename when given a path argument

### ISSUE THREAD

**Issue by oczkoisse** [PyCQA/pydocstyle#549]
As an example,

```
pydocstyle --match '(?!setup).*\.py' setup.py
```

correctly ignores `setup.py`. However, both

```
pydocstyle --match '(?!setup).*\.py' .\setup.py
```

and

```
pydocstyle --match '(?!setup).*\.py' ./setup.py
```

don't. This behavior is confusing for a user. Even worse, IDEs like VSCode automatially invoke `pydocstyle` with full path of opened Python file. Even if I include a `match` pattern in my configuration, it doesn't work because full paths are passed to `pydocstyle`. I wasted some time trying to figure out if I had configured `pydocstyle` correctly because VSCode was still flagging problems in ignored files.

I think the issue can be resolved by only considering basename of file arguments to match against `--match` pattern.

**Comment by sigmavirus24** [comment]
https://github.com/PyCQA/pydocstyle/blob/5a1c4d3a128ce662301a80532a578f37f8d5c3c3/src/pydocstyle/config.py#L243-L292

I think the solution can be handled in a couple ways:

1. We could switch from `re.compile(conf.match + '$').match` to `.search` at the end - this works because we are already adding a `$` to force the match to happen at the end.
2. We could leave that code as it is today and use `os.path.basename` on filenames (since `match-dir` should be handling the directory from the full-path)
3. On the odd chance someone is relying on this [behaviour](https://xkcd.com/1172/), add in `or match(os.path.basename(filename))` on line 280

https://docs.python.org/3/library/re.html#search-vs-match

**Comment by oczkoisse** [comment]
@sigmavirus24 Thanks for replying. About the suggested approaches:

1. It seems like changing to `.search` will break negative lookahead patterns because:
   ```python
   # The following evaluates to True with .search but is False with .match
   bool(re.compile('(?!test_).+\.py' + '$').search('test_a.py'))
   ```
   My understanding is that `pydocstyle` doesn't have an exclude option for files so people may be relying on negative lookahead patterns to get similar functionality.
2. I like this approach better because I think using `basename` is more explicit about `--match` being meant to match against filenames alone.
3. Personally, I think it might be okay to break the old behavior in this case. However, I leave that decision to project maintainers.
